lumicks/pylake/adjustments.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+
+
+class ColorAdjustment:
+    """Color adjustment for plotting"""
+
+    def __init__(self, minimum, maximum, mode="absolute"):
+        """Utility class to adjust the min/max values of image colormaps.
+
+        Values can be supplied as a single number or 1-component list, in which case the value is
+        applied to all color channels, or a 3-component list specifying the values for red, green,
+        and blue channels.
+
+        Parameters
+        ----------
+        minimum : array_like
+            1 or 3-component list of lower limits for the color mapping
+        maximum : array_like
+            1 or 3-component list of upper limits for the color mapping
+        mode : str
+            - "percentile" : color limits are given as percentiles of the image in question.
+              Percentiles are calculated for each color channel separately
+            - "absolute" : color limits are given as absolute values.
+            Note: When providing bounds in percentiles, limits will change depending on which image
+            you are looking at. When scrolling through a stack of images, the limits will not remain
+            constant.
+
+        Examples
+        --------
+        ::
+
+            from lumicks import pylake
+            file = pylake.File("example.h5")
+
+            # Plot scan with red color mapped from 5th to 95th percentile.
+            adjustment = lk.ColorAdjustment(5, 95, mode="percentile")
+            file.scans["scan"].plot(channel="red", adjustment=adjustment)
+
+            # Plot scan with RGB colors mapped from 5th to 95th percentile.
+            adjustment = lk.ColorAdjustment([5, 5, 5], [95, 95, 95], mode="percentile")
+            file.scans["scan"].plot(adjustment=adjustment)
+
+            stack = lk.CorrelatedStack("camera_recording.tiff")
+            # Plot force 1x with this stack and adjust the contrast by specifying an absolute upper
+            # and lower bound for the intensities.
+            absolute_adjustment = lk.ColorAdjustment([50, 30, 20], [1000, 195, 95])
+            stack.plot_correlated(file.force1x, channel="rgb", adjustment=absolute_adjustment)
+        """
+        minimum, maximum = np.atleast_1d(minimum), np.atleast_1d(maximum)
+        self.mode = mode
+        if self.mode == "nothing":
+            return
+
+        for bound in (minimum, maximum):
+            if len(bound) not in (1, 3):
+                raise ValueError("Color value bounds should be of length 1 or 3")
+
+        if mode not in ("percentile", "absolute", "nothing"):
+            raise ValueError("Mode nust be percentile or absolute")
```

Comment:
I guess we don't have to test for `"nothing"` here if we return above

Reply:
👍